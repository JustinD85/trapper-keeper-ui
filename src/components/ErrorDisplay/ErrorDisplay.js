import React from 'react';
import { withRouter, NavLink } from 'react-router-dom';
import { Button, Typography } from '@material-ui/core'

export const ErrorDisplay = (props) => {
  return (
    <div className='errorDisplay'>
      <Typography variant='display4'>Error 404</Typography>
      <Typography variant='display3'>Page not found.</Typography>
      <Typography variant='display2'>The path localhost:/3000{props.location.pathname} was not found.</Typography>
      <NavLink to='/'><Button>Return Home</Button></NavLink>
    </div>
  )
};

export default withRouter(ErrorDisplay);