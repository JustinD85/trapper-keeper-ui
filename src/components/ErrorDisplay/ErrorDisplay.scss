.errorDisplay {
  margin: 50px Auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}