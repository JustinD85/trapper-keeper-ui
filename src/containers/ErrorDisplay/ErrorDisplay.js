import React from "react";
import { withRouter } from "react-router-dom";

export const ErrorDisplay = (props) => {
  return (
    <div>
      <h2>Error 404</h2>
      <h3>Page not found.</h3>
      <p>The path localhost:/3000{props.location.pathname} was not found.</p>
    </div>
  )
};

export default withRouter(ErrorDisplay);