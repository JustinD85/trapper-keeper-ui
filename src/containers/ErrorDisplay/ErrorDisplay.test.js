import React from "react";
import { shallow } from "enzyme";
import { ErrorDisplay } from "./ErrorDisplay";

describe("ErrorDisplay", () => {
  let wrapper;
  it("should render with no error and match snapshot", () => {
    wrapper = shallow(<ErrorDisplay location='path.com' />);
    expect(wrapper).toMatchSnapshot();
  });
});
