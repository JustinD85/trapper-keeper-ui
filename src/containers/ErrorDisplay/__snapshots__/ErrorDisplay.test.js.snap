// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorDisplay should render with no error and match snapshot 1`] = `
<div>
  <h2>
    Error 404
  </h2>
  <h3>
    Page not found.
  </h3>
  <p>
    The path localhost:/3000
     was not found.
  </p>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/"
  >
    <button>
      Return Home
    </button>
  </NavLink>
</div>
`;
